File: school-system-front/src/types/rapport.ts
export type TypeRapport = "Bulletin trimestriel" | "Rapport annuel" | "Rapport de discipline" | "Rapport d'absence" | "Rapport financier" | "PV de conseil";

export type StatutRapport = "Brouillon" | "Généré" | "Validé" | "Envoyé";

export type Rapport = {
  id: string;
  titre: string;
  type: TypeRapport;
  periode: string;
  dateGeneration: string;
  statut: StatutRapport;
  auteur: string;
  destinataire: string;
  description: string;
  fichier: string;
};

export const TYPES_RAPPORT: TypeRapport[] = [
  "Bulletin trimestriel",
  "Rapport annuel",
  "Rapport de discipline",
  "Rapport d'absence",
  "Rapport financier",
  "PV de conseil",
];

export const STATUTS_RAPPORT: StatutRapport[] = [
  "Brouillon",
  "Généré",
  "Validé",
  "Envoyé",
];

File: school-system-back/src/main/java/com/schoolSys/schooolSys/rapport/Rapport.java
package com.schoolSys.schooolSys.rapport;

import java.util.UUID;
import org.hibernate.annotations.UuidGenerator;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "rapports")
@SQLDelete(sql = "UPDATE rapports SET deleted = true, deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted = false")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Rapport {

    @Id
    @GeneratedValue
    @UuidGenerator(style = UuidGenerator.Style.TIME)
    private UUID id;

    @Column(nullable = false, length = 255)
    private String titre;

    @Column(nullable = false, length = 100)
    private String type;

    @Column(length = 100)
    private String periode;

    @Column(name = "date_generation", nullable = false)
    @Builder.Default
    private LocalDateTime dateGeneration = LocalDateTime.now();

    @Column(nullable = false, length = 30)
    @Builder.Default
    private String statut = "Brouillon";

    @Column(nullable = false, length = 255)
    private String auteur;

    @Column(length = 255)
    private String destinataire;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 500)
    private String fichier;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(nullable = false)
    @Builder.Default
    private Boolean deleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
}
